processor/marshalprocessor/config.go

```diff
@@ -0,0 +1,61 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package marshalprocessor provides a processor that marshals logs to a specified format.
+package marshalprocessor
+
+import (
+	"errors"
+	"strings"
+
+	"go.uber.org/multierr"
+)
+
+var errInvalidMarshalTo = errors.New("marshal_to must be JSON, XML, or KV")
+var errXMLNotSupported = errors.New("XML not yet supported")
+var errKVSeparatorsEqual = errors.New("kv_separator and kv_pair_separator must be different")
+
+// Config is the configuration for the processor
+type Config struct {
+	MarshalTo       string `mapstructure:"marshal_to"` // MarshalTo is either JSON, XML, or KV
+	KVSeparator     rune   `mapstructure:"kv_separator"`
+	KVPairSeparator rune   `mapstructure:"kv_pair_separator"`
+}
+
+// Validate validates the processor configuration
+func (cfg Config) Validate() error {
+	var errs error
+
+	// Validate MarshalTo choice
+	switch strings.ToUpper(cfg.MarshalTo) {
+	case "JSON":
+	case "XML":
+		errs = multierr.Append(errs, errXMLNotSupported)
+	case "KV":
+		// Validate KV separators, which must be different from each other
+		if cfg.KVSeparator == cfg.KVPairSeparator && cfg.KVSeparator != 0 {
+			errs = multierr.Append(errs, errKVSeparatorsEqual)
+		}
+		if cfg.KVSeparator == 0 && cfg.KVPairSeparator == '=' {
+			errs = multierr.Append(errs, errKVSeparatorsEqual)
+		}
+		if cfg.KVPairSeparator == 0 && cfg.KVSeparator == ' ' {
+			errs = multierr.Append(errs, errKVSeparatorsEqual)
+		}
```

Comment:
I think this could be simplified by having the default config set KVPairSeparator and KVSeparator, then you'd only need the top condition.